[HumanBenchmark][Frontend][SignUpModal] add sign up modal

Benjamin-Snyder opened the pull request:
It is not implemented on the designated button yet

Does not take in a profile picture

Takes all other required user info and then creates the user in the database

Sorry I know its probably too long for a single PR, but wasnt sure how to break it up

jasonthaisy commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,266 @@
+      newErrors.age = "Please enter a valid number for age.";
+      valid = false;
+    }
+    if (!gender || (gender !== "M" && gender !== "F")) {
+      newErrors.gender = "Please enter 'M' or 'F' for gender.";
+      valid = false;
+    }
```
Why don't we use `<input type="radio">` to enforce choosing between M and F?

jasonthaisy commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,266 @@
+  const [weight, setWeight] = useState("");
+  const [age, setAge] = useState("");
+  const [gender, setGender] = useState("");
+  const [step, setStep] = useState(1);
```
Can we please create constants for the steps?

```
const STEP_ONE = 1;
const STEP_TWO = 2;
```

I know it seems redundant for this screen, but I really think we need to get into this habit. As I've mentioned before, skipping this type of code hygiene can result in pretty hard to solve bugs in large codebases, and its likely that other engineers will repeat this feedback in code reviews. Please address this before merging.

jasonthaisy commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,266 @@
+  const handleContinue = async (e) => {
+    e.preventDefault();
+
+    if (step === 1) {
```
see comment on line 14

jasonthaisy commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,266 @@
+      }
+
+      setStep(2);
+    } else if (step === 2) {
```
see comment on line 14

jasonthaisy commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,266 @@
+    return valid;
+  };
+
+  const handleContinue = async (e) => {
```
Prefer spelling out the full word when possible; `e -> event`

jasonthaisy commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,266 @@
+        try {
+            await createAccount(username, password, height, weight, age, gender);
+            // reset all inputs and return to step 1
+            setUsername("");
+            setPassword("");
+            setConfirmPassword("");
+            setHeight("");
+            setWeight("");
+            setAge("");
+            setGender("");
+            setErrors({
+              username: "",
+              password: "",
+              confirmPassword: "",
+              height: "",
+              weight: "",
+              age: "",
+              gender: "",
+            });
+            setStep(1);
```
Is it possible to wrap this into a helper function?

```
const resetState = () => {
  setUsername("");
  setPassword("");
  setConfirmPassword("");
  ...
}

jasonthaisy commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,266 @@
+              type="text"
+              id="username"
+              value={username}
+              onChange={(e) => setUsername(e.target.value)}
```
Prefer spelling out full word when possible `e -> event`

jasonthaisy reviewed (changes requested):
Let's also attach a screenshot or screen recording of the sign up modal to the PR since you've added it to the leaderboard page.

Benjamin-Snyder commented:
I just fixed the issues you brought up. 

change "e" to "event"
create helper function to reset modal
change gender input type to radio
created constants for step variables

I also added error handling for passwords under 10 characters

I tried to show off error handling in the video.
https://github.com/user-attachments/assets/6fcf8754-e8bb-442e-ab31-20b513f6ec72

Benjamin-Snyder commented:
added gender constants

lamjohnson commented on front-end/src/SignUpModal/SignUpModal.jsx:
```diff
@@ -0,0 +1,299 @@
+    }
+  };
+
+  const validateStep1 = () => {
```
you can be more specific on the naming eg. validateUserAccountSettings, validateUserPersonalInfo
